ai_transform/workflow/context_manager.py

```diff
@@ -90,85 +102,94 @@ def _set_field_children_recursively(self):
 
                     res = self.dataset[input_field].add_field_children(
                         field_children=output_fields,
-                        fieldchildren_id=self._job_id,
+                        fieldchildren_id=self.job_id,
                         metadata=metadata,
                         recursive=True,
                     )
                     logger.debug(format_logging_info(res))
 
+    def set_workflow_status(self, status: str):
+        result = self.api._set_workflow_status(
+            status=status,
+            job_id=self.job_id,
+            metadata=self.metadata,
+            workflow_name=self.workflow_name,
+            additional_information=self.additional_information,
+            send_email=self.send_email,
+            email=self.email,
+            worker_number=self.engine.worker_number,
+            output=self.engine.output_documents,
+        )
+        logger.debug(format_logging_info(result))
+        return result
+
+    def __enter__(self):
+        self._set_field_children_recursively()
+        return self.set_workflow_status(status=self.IN_PROGRESS)
+
     def _handle_workflow_fail(
         self, exc_type: type, exc_value: BaseException, traceback: Traceback
     ):
-        self._set_status(status=self.FAILED, worker_number=self._engine.worker_number)
         logger.exception(exc_value)
-        self._update_workflow_metadata(
-            job_id=self._job_id,
-            metadata={},
-        )
+        self.set_workflow_status(status=self.FAILED)
         return False
 
-    def _handle_workflow_complete(
-        self, exc_type: type, exc_value: BaseException, traceback: Traceback
-    ):
+    def trigger_polling_workflow(self):
```

Comment:
I don't think we have polling anymore - we should be able to just delete it but make sure that there's a **kwargs in the workflow function variable namespaces so that nothing breaks.

Reply:
Workflow(..., **kwargs)